cli/src/commands/help.rs

```diff
@@ -26,21 +30,94 @@ pub(crate) struct HelpArgs {
     pub(crate) command: Vec<String>,
 }
 
+struct Category {
+    description: &'static str,
+    content: &'static str,
+}
+
 #[instrument(skip_all)]
 pub(crate) fn cmd_help(
-    _ui: &mut Ui,
+    ui: &mut Ui,
     command: &CommandHelper,
     args: &HelpArgs,
 ) -> Result<(), CommandError> {
+    // TODO: Add all documentation to categories
+    //
+    // Maybe adding some code to build.rs to find all the docs files and build the
+    // HashMap at compile time.
+    //
+    // It would be cool to follow the docs hierarchy somehow.
+    //
+    // One of the problems would be `config.md`, as it has the same name as a
+    // subcommand.
+    let categories = BTreeMap::from([
+        (
+            "revsets",
+            Category {
+                description: "Documentation about revsets",
+                content: include_str!("../../../docs/revsets.md"),
+            },
+        ),
+        (
+            "tutorial",
+            Category {
+                description: "Show a tutorial to get started with jj",
+                content: include_str!("../../../docs/tutorial.md"),
+            },
+        ),
+    ]);
+
+    if let Some(category) = categories.get(args.command.join(" ").as_str()) {
+        ui.request_pager();
+        write!(ui.stdout(), "{}", category.content)?;
+
+        return Ok(());
+    }
+
     let mut args_to_show_help = vec![command.app().get_name()];
     args_to_show_help.extend(args.command.iter().map(|s| s.as_str()));
     args_to_show_help.push("--help");
 
+    let help_template_before_categories = "\
+    {about-with-newline}\n{usage-heading} {usage}\n\n\x1b[1m\x1b[4mCommands:\x1b[0m\n{subcommands}
+    ";
+
+    let subcommand_max_len = command
+        .app()
+        .get_subcommands()
+        .map(|cmd| cmd.get_name().len())
+        .max()
+        .unwrap();
+
+    let help_template_categories = format!(
+        "\x1b[1m\x1b[4mHelp Categories:\x1b[0m\n{}",
+        &categories
+            .iter()
+            .map(|(&name, category)| format!(
+                "{{tab}}\x1b[1m{:<width$}\x1b[0m{{tab}}{}",
+                name,
+                category.description,
+                width = subcommand_max_len
+            ))
+            .join("\n")
+    );
```

Comment:
nit: do category stuff only if `args.command.is_empty()`? With the current implementation, it's not obvious that the categories list is inserted only to the top-level help.

Reply:
With my testing, it seems to work only on the first level, but I couldn't find a statement saying that it will, or will not, apply it recursively through the subcommands.

Reply:
I don't think the template option applies recursively, but it's clearer if we do that explicitly.

Anyway, I would remove all templating stuff and just use `after_long_help`.